### PR TITLE
Fix for 32-bit "uU" dateformat issue

### LINKED ISSUE #187: parse(ZonedDateTime, date_str, dateformat"someformat") fails on 32-bit linux
Using 0.8.5

```julia
julia> using Dates, TimeZones

julia> zt = parse(ZonedDateTime, "Feb 5 00:00:00 2019 GMT", dateformat"u d H:M:S Y Z")
ERROR: MethodError: no method matching ZonedDateTime(::Int64, ::Int32, ::Int64, ::Int64, ::Int64, ::Int64, ::Int64, ::SubString{String})
Closest candidates are:
  ZonedDateTime(::Int64, ::Int64, ::Int64, ::Int64, ::Int64, ::Int64, ::Int64, ::AbstractString) at /root/.julia/packages/TimeZones/WMDpl/src/types.jl:252
  ZonedDateTime(::Integer, ::Integer, ::Integer, ::Integer, ::Integer, ::Integer, ::Integer, ::VariableTimeZone, ::Integer) at /root/.julia/packages/TimeZones/WMDpl/src/types.jl:242
  ZonedDateTime(::Integer, ::Integer, ::Integer, ::Integer, ::Integer, ::Integer, ::Integer, ::TimeZone) at /root/.julia/packages/TimeZones/WMDpl/src/types.jl:246
  ...
Stacktrace:
 [1] parse(::Type{ZonedDateTime}, ::String, ::DateFormat{Symbol("u d H:M:S Y Z"),Tuple{Dates.DatePart{'u'},Dates.Delim{Char,1},Dates.DatePart{'d'},Dates.Delim{Char,1},Dates.DatePart{'H'},Dates.Delim{Char,1},Dates.DatePart{'M'},Dates.Delim{Char,1},Dates.DatePart{'S'},Dates.Delim{Char,1},Dates.DatePart{'Y'},Dates.Delim{Char,1},Dates.DatePart{'Z'}}}) at /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.0/Dates/src/parse.jl:285
 [2] top-level scope at none:0
```
Looks like https://github.com/JuliaTime/TimeZones.jl/blob/v0.8.5/src/types.jl#L251 expects all the arguments to be Int64, but on a 32-bit system that's not the case.

### PR DESCRIPTION
**Pull request by fchorney**
closes #187

### CODE REVIEW

File: src/types/zoneddatetime.jl [hunk 1]
```diff
@@ -118,7 +118,8 @@ end
 
 # Parsing constructor. Note we typically don't support passing in time zone information as a
 # string since we cannot do not know if we need to support resolving ambiguity.
-function ZonedDateTime(y::Int64, m::Int64, d::Int64, h::Int64, mi::Int64, s::Int64, ms::Int64, tz::AbstractString)
+# Month Union issue: https://github.com/JuliaTime/TimeZones.jl/issues/187#issuecomment-473012078
+function ZonedDateTime(y::Int64, m::Union{Int32, Int64}, d::Int64, h::Int64, mi::Int64, s::Int64, ms::Int64, tz::AbstractString)
```
Comment (omus) [inline src/types/zoneddatetime.jl]: Can we add a test for this? AppVeyor should run on 32-bit.

Comment (stevengj) [reply src/types/zoneddatetime.jl]: Wouldn't it be better to just declare the arguments as `Integer`?   The `DateTime` constructor will convert them as needed, and the compiler will type-specialize the values anyway.

Comment (omus) [reply src/types/zoneddatetime.jl]: I'd honestly prefer not to have this constructor (see the note in the code). Keeping the types very tight here discourages general usage. The only reason this constructor exists is for interoperability with the Dates parsing code.

**Comment by omus** [comment]
AppVeyor is failing on 32-bit nightly on trying to download Julia

**Pull request merged**